braindecode/models/modules.py

```diff
@@ -528,102 +529,289 @@ def extra_repr(self) -> str:
+                UserWarning,
+            )
+            low_freq = 4
+            high_freq = 40
+
+            band_filters = self._create_band_filters(low_freq, high_freq, band_filters)
+        if not isinstance(band_filters, list):
+            ValueError(
+                "`band_filters` should be a list of tuples if you want to "
+                "use them this way."
+            )
 
-        self.orig_freq = orig_freq
-        self.new_freq = new_freq
-        self.gcd = math.gcd(int(self.orig_freq), int(self.new_freq))
-        self.resampling_method = resampling_method
-        self.lowpass_filter_width = lowpass_filter_width
-        self.rolloff = rolloff
-        self.beta = beta
-
-        if self.orig_freq != self.new_freq:
-            kernel, self.width = _get_sinc_resample_kernel(
-                self.orig_freq,
-                self.new_freq,
-                self.gcd,
-                self.lowpass_filter_width,
-                self.rolloff,
-                self.resampling_method,
-                beta,
-                dtype=dtype,
+        # and we accepted as
+        self.band_filters = band_filters
+        self.n_bands = len(band_filters)
+        self.phase = phase
+        self.method = method
+        self.n_chans = n_chans
+
+        self.method_iir = True if self.method == "iir" else False
+
+        if self.method_iir:
+            if iir_params is None:
+                iir_params = dict(output="ba")
+            else:
+                if "output" in iir_params:
+                    if iir_params["output"] == "sos":
+                        warn("""
+                            It is not possible to use second-order section
+                            filtering with Torch. Changing to filter ``ba``
+                            """)
+                        iir_params["output"] = "ba"
+
+        filts = {}
+        for idx, (l_freq, h_freq) in enumerate(band_filters):
+            filt = create_filter(
+                data=None,
+                sfreq=sfreq,
+                l_freq=l_freq,
+                h_freq=h_freq,
+                filter_length=filter_length,
+                l_trans_bandwidth=l_trans_bandwidth,
+                h_trans_bandwidth=h_trans_bandwidth,
+                method=method,
+                iir_params=iir_params,
+                phase=phase,
+                fir_window=fir_window,
+                fir_design=fir_design,
+                verbose=verbose,
             )
-            self.register_buffer("kernel", kernel)
+            if not self.method_iir:
+                filts[f"{idx}_band"] = from_numpy(filt).float()
+            else:
+                b = from_numpy(filt["b"]).float()
+                a = from_numpy(filt["a"]).float()
+
+                filts[f"{idx}_band"] = (b, a)
 
-    def forward(self, waveform: Tensor) -> Tensor:
-        r"""
-        Args:
-            waveform (Tensor): Tensor of audio of dimension (..., time).
+            self.filts = filts
```

Comment:
If you want to be able to move to GPU, you need to store this as
```suggestion
            self.filts = nn.ModuleDict(filts)
```

Reply:
Now store at the GPU